src/algebra/group/hom.lean

```diff
@@ -1,18 +1,50 @@
 /-
-Copyright (c) 2014 Patrick Massot. All rights reserved.
+Copyright (c) 2018 Patrick Massot. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Patrick Massot, Kevin Buzzard, Scott Morrison, Johan Commelin, Chris Hughes, Johannes Hölzl, Yury Kudryashov
 
 Homomorphisms of multiplicative and additive (semi)groups and monoids.
+
+explain why no group_hom G H
 -/
+
 import algebra.group.to_additive algebra.group.basic
 
+/-!
+# monoid and group homomorphisms
+
+This file defines the basic structures for monoid and group
+homomorphisms, both unbundled (e.g. `is_group_hom f`) and bundled
+(e.g. `group_hom f`). The unbundled ones are deprecated and the plan is
+to slowly remove them from mathlib.
+
+## main definitions
+
+moniod_hom, is_monoid_hom, group_hom. is_group_hom
+
+## Notations
+
+→* for bundled monoid homs (also use for group homs)
+→+ for bundled add_monoid homs (also use for add_group homs)
+
+## implementation notes
+
+There's a coercion from bundled homs to fun, and the canonical
+notation is to use the bundled hom as a function via this coercion.
+
+## Tags
+
+is_group_hom, group_hom, is_monoid_hom, monoid_hom
+
+-/
 universes u v
 variables {α : Type u} {β : Type v}
 
+/-- predicate for maps which preserve a multiplication -/
```

Comment:
I think (but I'm not sure) that the style is to capitalize the first word in a docstring. (And maybe also a period at the end.)

Reply:
OK!